data/model/oci/label.py

```diff
@@ -84,6 +102,44 @@ def create_manifest_label(manifest_id, key, value, source_type_name, media_type_
         raise InvalidMediaTypeException()
 
     source_type_id = Label.source_type.get_id(source_type_name)
+    return dict(key=key, value=value, source_type=source_type_id, media_type=media_type_id)
+
+
+def batch_create_manifest_labels(manifest_id, labels):
+    """ Creates labels on a manifest in a batch fashion. The labels argument
+        should be an iterator or list containing dict's whose values matches the
+        arguments to create_manifest_label (minus the manifest_id)
+    """
+    translated = [_check_and_translate_label(**label) for label in labels]
+
+    # Ensure the manifest exists.
+    try:
+        manifest = Manifest.select(Manifest).where(Manifest.id == manifest_id).get()
+    except Manifest.DoesNotExist:
+        return None
+
+    label_ids = list()
+    with db_transaction():
+        # Create the labels. We do this non-batch-wise because we need the returned IDs.
```

Comment:
Heh, postgres actually has a `RETURNING` clause to return data on bulk inserts.

Reply:
I wish we could use it :/